CFC_WebApp/clients/socialmedia/settings.json.sample

```diff
@@ -0,0 +1,4 @@
+{
+    "fb_app_secret":"ask sid",
```

Comment:
Yeah. I think having a service-specific settings file makes sense if you are sensitive to the format of the settings.json file. However, my personal preference is externalizing settings as development environment variables. 

I had some interest in using this for a previous project: http://yamjam.readthedocs.org/en/latest/index.html <http://yamjam.readthedocs.org/en/latest/index.html>, which is not just django-specific. This hoists the configuration variables out of anything under source control while implementing a reasonably simple API.   

It may be worth the investment of time now to have a central config structure and avoid sprinkling config files in various locations and having to manage the risk of inadvertent check-in later. Perhaps there is some sense to this that I am unfamiliar with; however. 

Sid Feygin
sid.feygin@gmail.com



> On Jun 2, 2015, at 11:17 PM, shankari <notifications@github.com> wrote:
> 
> In CFC_WebApp/clients/socialmedia/settings.json.sample <https://github.com/e-mission/e-mission-server/pull/133#discussion_r31596328>:
> 
> > @@ -0,0 +1,4 @@
> > +{
> > +    "fb_app_secret":"ask sid",
> I think that it is fine to leave it as settings.json.sample. My main comments are:
> 
> Maybe we want to have separate settings for each external service (e.g. facebook_settings.json.sample). I am not too insistent on this and am willing to accept your current implementation (settings.json.sample with service specific prefixes, e.g. fb_app_secret).
> The format of settings.json.sample should be the same as the format of settings.json. Otherwise, people will replace the keys and then spend a bunch of time trying to figure out why they are getting parse errors. In your case, for example, in settings.json, fb_app_secret is a child of client_settings while in settings.json.sample, it is a top level element.
> —
> Reply to this email directly or view it on GitHub <https://github.com/e-mission/e-mission-server/pull/133/files#r31596328>.
>